pandas/tests/tseries/offsets/test_ticks.py

```diff
@@ -319,26 +325,32 @@ def test_compare_ticks(cls):
     assert cls(3) != cls(4)
 
 
-@pytest.mark.parametrize("cls", tick_classes)
-def test_compare_ticks_to_strs(cls):
+@pytest.mark.parametrize(
+    "cls, left, right, comparison",
+    [
+        (cls, left, right, comparison)
+        for cls in tick_classes
```

Comment:
We don't want for loops inside the parameters, parametrizing is already a type of for loop. If you check some of the other examples in the code base it might become a bit more clear.

Reply:
i will have look at this. I used this as left, right arguments were dependent on cls vlaue.

Reply:
Ah I missed that part, maybe parametrizing isn't as clean as I thought

Reply:
so should i remove it from left and right. And keep it only for cls and comparison.

Reply:
For now I suppose just adding that error message should be good for this PR (any other cleanup could always be done as follow-up). Sorry for the confusion.